pkg/io/video/scaler_cgo.go

```diff
@@ -92,3 +110,74 @@ func (f *FastNearestNeighbor) Scale(dst draw.Image, dr image.Rectangle, src imag
 		panic("unimplemented")
 	}
 }
+
+// FastBoxSampling is a CGO version of Box sampling scaler.
+type FastBoxSampling struct {
+}
+
+// Scale implements the draw.Scaler interface.
+func (f *FastBoxSampling) Scale(dst draw.Image, dr image.Rectangle, src image.Image, sr image.Rectangle, op draw.Op, opts *draw.Options) {
+	ddx, ddy := dr.Dx(), dr.Dy()
+	sdx, sdy := sr.Dx(), sr.Dy()
+	if sdx < ddx && sdy < ddy {
+		// Box upsampling is equivalent of NearestNeighbor
+		(&FastNearestNeighbor{}).Scale(dst, dr, src, sr, op, opts)
+		return
+	}
+	tmp := poolBoxSampleBuffer.Get().(*[]uint32)
+	defer poolBoxSampleBuffer.Put(tmp)
```

Comment:
We can have two `MediaStream`s to concurrently use two cameras, so the buffer can't be a global buffer.
Also, since `Scaler` instance is shared in the process as it is defined like https://github.com/golang/image/blob/58c23975cae11f062d4b3b0c143fe248faac195d/draw/scale.go#L154-L158, it can't be a member of scaler struct.

Reply:
Ah, that makes sense.